code/datums/components/trader/trader.dm

```diff
@@ -0,0 +1,413 @@
+	if(!can_trade(customer))
+		return
+	if(npc_result != "Yes")
+		trader.say(trader_data.return_trader_phrase(ITEM_SELLING_CANCELED_PHRASE))
+		return TRUE
+	trader.say(trader_data.return_trader_phrase(ITEM_SELLING_ACCEPTED_PHRASE))
+	playsound(trader, trader_data.sell_sound, 50, TRUE)
+	log_econ("[selling] has been sold to [trader] (typepath used for product info; [typepath_for_product_info]) by [customer] for [cost] cash.")
+	exchange_sold_items(selling, cost, typepath_for_product_info)
+	generate_cash(cost, customer)
+	return TRUE
+
+/**
+ * Modifies the 'base' price of a item based on certain variables
+ *
+ * Arguments:
+ * * Reference to the item; this is the item being sold
+ * * Original cost; the original cost of the item, to be manipulated depending on the variables of the item, one example is using item.amount if it's a stack
+ */
+/datum/component/trader/proc/apply_sell_price_mods(obj/item/selling, original_cost)
+	if(isstack(selling))
+		var/obj/item/stack/stackoverflow = selling
+		original_cost *= stackoverflow.amount
+	return original_cost
+
+/**
+ * Handles modifying/deleting the items to ensure that a proper amount is converted into cash; put into it's own proc to make the children of this not override a 30+ line sell_item()
+ *
+ * Arguments:
+ * * selling - (Item REF) this is the item being sold
+ * * value_exchanged_for - (Number) the "value", useful for a scenario where you want to remove enough items equal to the value
+ * * original_typepath - (Typepath) For scenarios where a children of a parent is being sold but we want to modify the parent's product information
+ */
+/datum/component/trader/proc/exchange_sold_items(obj/item/selling, value_exchanged_for, original_typepath)
+	var/list/product_info = wanted_items[original_typepath]
+	if(isstack(selling))
+		var/obj/item/stack/the_stack = selling
+		var/actually_sold = min(the_stack.amount, product_info[TRADER_PRODUCT_INFO_QUANTITY])
+		the_stack.use(actually_sold)
+		product_info[TRADER_PRODUCT_INFO_QUANTITY] -= (actually_sold)
+	else
+		qdel(selling)
+		product_info[TRADER_PRODUCT_INFO_QUANTITY] -= 1
+
+/**
+ * Creates an item equal to the value set by the proc and puts it in the user's hands if possible
+ * Arguments:
+ * * value - A number; The amount of cash that will be on the holochip
+ * * customer - Reference to a mob; The mob we put the holochip in hands of
+ */
+/datum/component/trader/proc/generate_cash(value, mob/customer)
+	var/obj/item/holochip/chip = new /obj/item/holochip(get_turf(customer), value)
+	customer.put_in_hands(chip)
+
+///Talk about what items are being sold/wanted by the trader and in what quantity or lore
+/datum/component/trader/proc/discuss(mob/customer)
+	var/mob/living/trader = parent
+	var/list/npc_options = list(
+		"Lore" = radial_icons_cache[TRADER_RADIAL_LORE],
+		"Selling?" = radial_icons_cache[TRADER_RADIAL_DISCUSS_SELL],
+		"Buying?" = radial_icons_cache[TRADER_RADIAL_DISCUSS_BUY],
```

Comment:
More possible define candidates, which could trip people up on the basis of only two of them using `?`